### PR TITLE
Add bulk operations for ByteBuffer

### PR DESCRIPTION
**Pull request by normanmaurer**
Motivation:

We should provide bulk operations that either write a ByteBuffer or read into a ByteBuffer

Modifications:

- Add implementation
- Add unit tests

Result:

Expose bulk operations for ByteBuffer

### CODE REVIEW

File: buffer/src/main/java/io/netty5/buffer/api/Buffer.java [hunk 1]
```diff
@@ -399,6 +400,70 @@ default Buffer writeBytes(byte[] source, int srcPos, int length) {
...
+    default Buffer writeBytes(ByteBuffer source) {
+        if (source.hasArray()) {
+            writeBytes(source.array(), source.arrayOffset() + source.position(), source.remaining());
+            source.position(source.limit());
+        } else {
+            int woff = writerOffset();
+            int length = source.remaining();
+            writerOffset(woff + source.remaining());
+            // Try to reduce bounce-checking by using long and int when possible.
+            boolean needReverse = source.order() != ByteOrder.BIG_ENDIAN;
```
Comment (chrisvest) [inline buffer/src/main/java/io/netty5/buffer/api/Buffer.java:420]: You can also temporarily change the order of the source to match the sink.

File: buffer/src/main/java/io/netty5/buffer/api/Buffer.java [hunk 2]
```diff
@@ -399,6 +400,70 @@ default Buffer writeBytes(byte[] source, int srcPos, int length) {
         return this;
     }
 
+    /**
+     * Writes into this buffer from the source {@link ByteBuffer}.
+     * This updates the {@linkplain #writerOffset() write offset} of this buffer and also the position of
+     * the source {@link ByteBuffer}.
+     *
```
Comment (chrisvest) [inline buffer/src/main/java/io/netty5/buffer/api/Buffer.java:407]: We can also call out that the behaviour is undefined if the given `ByteBuffer` is an alias of this `Buffer`.

File: buffer/src/main/java/io/netty5/buffer/api/Buffer.java [hunk 3]
```diff
@@ -399,6 +400,70 @@ default Buffer writeBytes(byte[] source, int srcPos, int length) {
...
+
+    /**
+     * Read from this buffer, into the destination {@link ByteBuffer}
+     * This updates the {@linkplain #readerOffset() read offset} of this buffer and also the position of
+     * the destination {@link ByteBuffer}.
+     *
+     * @param destination The {@link ByteBuffer} to write into.
+     * @return This buffer.
+     */
+    default Buffer readBytes(ByteBuffer destination) {
```
Comment (chrisvest) [inline buffer/src/main/java/io/netty5/buffer/api/Buffer.java:442]: The read version can be implemented in terms of `copyInto`

**Review by chrisvest — COMMENTED** [review]
Looks like useful additions. Some comments.

File: buffer/src/main/java/io/netty5/buffer/api/Buffer.java (hunk 1 above)
Comment (normanmaurer) [reply buffer/src/main/java/io/netty5/buffer/api/Buffer.java:420]: I thought about this but I think this one should be faster and produce less GC.

**Comment by chrisvest** [comment]
@normanmaurer I added my suggested changes. Merging it assuming you agree, otherwise we can take it in a follow up.

**Pull request merged**